components/eventing-controller/controllers/subscription/jetstream/reconciler_test.go

```diff
@@ -468,6 +468,36 @@ func TestChangeSubscription(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "CleanEventTypes; update valid filter to a filter with an invalid prefix",
+			givenSubscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter(reconcilertesting.EventSource, utils.NewCleanEventType("0")),
+				reconcilertesting.WithWebhookForNATS(),
+				reconcilertesting.WithSinkURLFromSvc(ens.SubscriberSvc),
+			},
+			wantBefore: utils.Want{
+				K8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultReadyCondition()),
+					reconcilertesting.HaveSubsConfiguration(utils.ConfigDefault(ens.DefaultSubscriptionConfig.MaxInFlightMessages)),
+					reconcilertesting.HaveCleanEventTypes([]string{
+						utils.NewCleanEventType("0"),
+					}),
+				},
+			},
+			changeSubscription: func(subscription *eventingv1alpha1.Subscription) {
+				// change the filter  to the event type
+				for _, f := range subscription.Spec.Filter.Filters {
+					f.EventType.Value = fmt.Sprintf("invalid%s", f.EventType.Value)
+				}
+			},
+			wantAfter: utils.Want{
+				K8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveConditionInvalidPrefix(),
+					reconcilertesting.HaveSubsConfiguration(utils.ConfigDefault(ens.DefaultSubscriptionConfig.MaxInFlightMessages)),
+					reconcilertesting.HaveCleanEventTypes(nil),
```

Comment:
nit^10: Here is a small suggestion to further improve the readability of the code. Feel free to ignore it if you don't like it :) 
The idea is to have another function to make clear that `nil` means **empty CleanEventTypes**

```suggestion
					reconcilertesting.HaveCleanEventTypesEmpty(),
```

where
```go
func HaveCleanEventTypesEmpty() gomegatypes.GomegaMatcher {
	return HaveCleanEventTypes(nil)
}
```

Reply:
I don't have a strong opinion here to be honest. But I think it's slightly better than it is right now. So I will do implement it

Reply:
Done